Panic on optional TLS settings

<!-- Thank you for your contribution. Before you submit the issue:
1. Search open and closed issues for duplicates.
2. Read the contributing guidelines.
-->

**Description**

With current version 1.18.0 of telemetry-manager a panic is observed on metricpipelines having no TLS settings defined:

```
panic: runtime error: invalid memory address or nil pointer dereference
	panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1630bb8]

goroutine 601 [running]:
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile.func1()
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.18.4/pkg/internal/controller/controller.go:111 +0x1e5
panic({0x18312c0?, 0x2a077e0?})
	/usr/local/go/src/runtime/panic.go:770 +0x132
github.com/kyma-project/telemetry-manager/internal/tlscert.(*Validator).Validate(0xc000513080, {0x1d08298, 0xc001683920}, {0xc001c02108?, 0xc001c02120?, 0x0?})
	/telemetry-manager-workspace/internal/tlscert/tls_cert_validator.go:112 +0x118
github.com/kyma-project/telemetry-manager/internal/reconciler/logpipeline.(*Reconciler).evaluateConfigGeneratedCondition(0xc0002a9708, {0x1d08298, 0xc001683920}, 0xc00189c000)
	/telemetry-manager-workspace/internal/reconciler/logpipeline/status.go:119 +0x1e4
github.com/kyma-project/telemetry-manager/internal/reconciler/logpipeline.(*Reconciler).setFluentBitConfigGeneratedCondition(0x0?, {0x1d08298?, 0xc001683920?}, 0xc00189c000)
	/telemetry-manager-workspace/internal/reconciler/logpipeline/status.go:90 +0x37
github.com/kyma-project/telemetry-manager/internal/reconciler/logpipeline.(*Reconciler).updateStatus(0xc0002a9708, {0x1d08298, 0xc001683920}, {0xc0007658dc, 0x3})
	/telemetry-manager-workspace/internal/reconciler/logpipeline/status.go:41 +0x139
github.com/kyma-project/telemetry-manager/internal/reconciler/logpipeline.(*Reconciler).doReconcile.func1()
	/telemetry-manager-workspace/internal/reconciler/logpipeline/reconciler.go:143 +0x38
panic({0x18312c0?, 0x2a077e0?})
	/usr/local/go/src/runtime/panic.go:770 +0x132
github.com/kyma-project/telemetry-manager/internal/tlscert.(*Validator).Validate(0xc000513080, {0x1d08298, 0xc001683920}, {0xc001d5e270?, 0xc001d5e288?, 0x0?})
	/telemetry-manager-workspace/internal/tlscert/tls_cert_validator.go:112 +0x118
github.com/kyma-project/telemetry-manager/internal/reconciler/logpipeline.(*Reconciler).isReconcilable(0xc0002a9708, {0x1d08298, 0xc001683920}, 0xc0002fc008)
	/telemetry-manager-workspace/internal/reconciler/logpipeline/reconciler.go:322 +0x16d
github.com/kyma-project/telemetry-manager/internal/reconciler/logpipeline.(*Reconciler).getReconcilablePipelines(0xc0002a9708, {0x1d08298, 0xc001683920}, {0xc0002fc008, 0x2, 0x0?})
	/telemetry-manager-workspace/internal/reconciler/logpipeline/reconciler.go:295 +0xbc
github.com/kyma-project/telemetry-manager/internal/reconciler/logpipeline.(*Reconciler).doReconcile(0xc0002a9708, {0x1d08298, 0xc001683920}, 0xc001685a00)
	/telemetry-manager-workspace/internal/reconciler/logpipeline/reconciler.go:165 +0x29a
github.com/kyma-project/telemetry-manager/internal/reconciler/logpipeline.(*Reconciler).Reconcile(0xc0002a9708, {0x1d08298, 0xc001683920}, {{{0x0?, 0x0?}, {0xc0007658dc?, 0x5?}}})
	/telemetry-manager-workspace/internal/reconciler/logpipeline/reconciler.go:137 +0x1a5
github.com/kyma-project/telemetry-manager/controllers/telemetry.(*LogPipelineController).Reconcile(0xc000d62c00?, {0x1d08298?, 0xc001683920?}, {{{0x0?, 0x5?}, {0xc0007658dc?, 0xc0007c9d10?}}})
	/telemetry-manager-workspace/controllers/telemetry/logpipeline_controller.go:53 +0x2b
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0x1d0c178?, {0x1d08298?, 0xc001683920?}, {{{0x0?, 0xb?}, {0xc0007658dc?, 0x0?}}})
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.18.4/pkg/internal/controller/controller.go:114 +0xb7
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc00035a370, {0x1d082d0, 0xc00051b040}, {0x18b9180, 0xc000518420})
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.18.4/pkg/internal/controller/controller.go:311 +0x3bc
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc00035a370, {0x1d082d0, 0xc00051b040})
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.18.4/pkg/internal/controller/controller.go:261 +0x1be
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2()
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.18.4/pkg/internal/controller/controller.go:222 +0x79
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2 in goroutine 116
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.18.4/pkg/internal/controller/controller.go:218 +0x486
```

```yaml
spec:
  input:
    istio:
      enabled: false
    prometheus:
      enabled: false
      namespaces:
        exclude:
        - kyma-system
        - kube-system
        - istio-system
        - compass-system
    runtime:
      enabled: true
      namespaces:
        include:
        - cap
  output:
    otlp:
      endpoint:
        value: http://my-destination.my-namespace.svc.cluster.local:4317
      protocol: grpc
```

<!-- Provide a clear and concise description of the problem.
Describe where it appears, when it occurred, and what it affects.  -->

<!-- Provide relevant technical details such as the browser name and version, or the operating system. -->

**Expected result**
No crashloop
<!-- Describe what you expect to happen. -->

**Actual result**
A crashloop of manager pod
<!-- Describe what happens instead. -->

**Steps to reproduce**

<!-- List the steps to follow to reproduce the bug. Attach any files, links, code samples, or screenshots that could help in investigating the problem. -->

**Troubleshooting**

<!-- Describe the steps you have already taken to solve the issue. -->

**Release Notes**

```

```

--- a/internal/tlscert/tls_cert_validator.go
+++ b/internal/tlscert/tls_cert_validator.go
@@ -109,17 +109,9 @@ func (v *Validator) Validate(ctx context.Context, config TLSBundle) error {
 		return err
 	}
 
-	// CA is optional (check if missing)
-	missingCA := false
-	caPEM, err := resolveValue(ctx, v.client, *config.CA)
-	if err != nil {
-		missingCA = true
-	}
-
-	// Make the best effort replacement of linebreaks in cert/key if present.
+	// Make the best effort to replace linebreaks in cert/key if present.
 	sanitizedCert := bytes.ReplaceAll(certPEM, []byte("\\n"), []byte("\n"))
 	sanitizedKey := bytes.ReplaceAll(keyPEM, []byte("\\n"), []byte("\n"))
-	sanitizedCA := bytes.ReplaceAll(caPEM, []byte("\\n"), []byte("\n"))
 
 	// Parse the certificate
 	parsedCert, err := parseCertificate(sanitizedCert, ErrCertDecodeFailed, ErrCertParseFailed)
@@ -137,12 +129,20 @@ func (v *Validator) Validate(ctx context.Context, config TLSBundle) error {
 		return err
 	}
 
-	// CA is optional (don't parse and validate if missing)
-	if missingCA {
+	// CA is optional (don't parse, sanitize, and validate if missing)
+	if config.CA == nil {
 		return nil
 	}
 
-	// Parse the CA
+	caPEM, err := resolveValue(ctx, v.client, *config.CA)
+	if err != nil {
+		return nil
+	}
+
+	// Make the best effort to replace linebreaks in CA if present.
+	sanitizedCA := bytes.ReplaceAll(caPEM, []byte("\\n"), []byte("\n"))
+
+	// Parse the CA(s)
 	parsedCAs, err := parseCertificates(sanitizedCA, ErrCADecodeFailed, ErrCAParseFailed)
 	if err != nil {
 		return err

--- a/internal/tlscert/tls_cert_validator.go
+++ b/internal/tlscert/tls_cert_validator.go
@@ -136,7 +136,7 @@ func (v *Validator) Validate(ctx context.Context, config TLSBundle) error {
 
 	caPEM, err := resolveValue(ctx, v.client, *config.CA)
 	if err != nil {
-		return nil
+		return nil //nolint:nilerr // ErrValueResolveFailed irrelevant since CA is optional
 	}
 
 	// Make the best effort to replace linebreaks in CA if present.

--- a/internal/tlscert/tls_cert_validator.go
+++ b/internal/tlscert/tls_cert_validator.go
@@ -109,9 +109,21 @@ func (v *Validator) Validate(ctx context.Context, config TLSBundle) error {
 		return err
 	}
 
-	// Make the best effort to replace linebreaks in cert/key if present.
+	// CA is optional (check if missing)
+	missingCA := config.CA == nil
+
+	var caPEM []byte
+	if !missingCA {
+		caPEM, err = resolveValue(ctx, v.client, *config.CA)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Make the best effort to replace linebreaks in cert/key/ca if present.
 	sanitizedCert := bytes.ReplaceAll(certPEM, []byte("\\n"), []byte("\n"))
 	sanitizedKey := bytes.ReplaceAll(keyPEM, []byte("\\n"), []byte("\n"))
+	sanitizedCA := bytes.ReplaceAll(caPEM, []byte("\\n"), []byte("\n"))
 
 	// Parse the certificate
 	parsedCert, err := parseCertificate(sanitizedCert, ErrCertDecodeFailed, ErrCertParseFailed)
@@ -129,18 +141,10 @@ func (v *Validator) Validate(ctx context.Context, config TLSBundle) error {
 		return err
 	}
 
-	// CA is optional (don't parse, sanitize, and validate if missing)
-	if config.CA == nil {
+	// CA is optional (don't parse and validate if missing)
+	if missingCA {
 		return nil
 	}
-
-	caPEM, err := resolveValue(ctx, v.client, *config.CA)
-	if err != nil {
-		return nil //nolint:nilerr // ErrValueResolveFailed irrelevant since CA is optional
-	}
-
-	// Make the best effort to replace linebreaks in CA if present.
-	sanitizedCA := bytes.ReplaceAll(caPEM, []byte("\\n"), []byte("\n"))
 
 	// Parse the CA(s)
 	parsedCAs, err := parseCertificates(sanitizedCA, ErrCADecodeFailed, ErrCAParseFailed)

--- a/internal/tlscert/tls_cert_validator.go
+++ b/internal/tlscert/tls_cert_validator.go
@@ -99,17 +99,22 @@ func New(client client.Client) *Validator {
 }
 
 func (v *Validator) Validate(ctx context.Context, config TLSBundle) error {
-	certPEM, err := resolveValue(ctx, v.client, *config.Cert, ErrValueResolveFailed)
+	certPEM, err := resolveValue(ctx, v.client, *config.Cert)
 	if err != nil {
 		return err
 	}
 
-	keyPEM, err := resolveValue(ctx, v.client, *config.Key, ErrValueResolveFailed)
+	keyPEM, err := resolveValue(ctx, v.client, *config.Key)
 	if err != nil {
 		return err
 	}
 
-	caPEM, _ := resolveValue(ctx, v.client, *config.CA, nil)
+	// CA is optional
+	missingCA := false
+	caPEM, err := resolveValue(ctx, v.client, *config.CA)
+	if err != nil {
+		missingCA = true
+	}
 
 	// Make the best effort replacement of linebreaks in cert/key if present.
 	sanitizedCert := bytes.ReplaceAll(certPEM, []byte("\\n"), []byte("\n"))
@@ -129,7 +134,7 @@ func (v *Validator) Validate(ctx context.Context, config TLSBundle) error {
 
 	// Parse the CA
 	var parsedCAs []*x509.Certificate
-	if caPEM != nil {
+	if !missingCA {
 		parsedCAs, err = parseCertificates(sanitizedCA, ErrCADecodeFailed, ErrCAParseFailed)
 		if err != nil {
 			return err
@@ -232,24 +237,7 @@ func validateCA(ca *x509.Certificate, now time.Time) error {
 	return nil
 }
 
-// func resolveValue(ctx context.Context, c client.Reader, value telemetryv1alpha1.ValueType, errResolve error) ([]byte, error) {
-// 	if value.Value != "" {
-// 		return []byte(value.Value), nil
-// 	}
-
-// 	if value.ValueFrom == nil || !value.ValueFrom.IsSecretKeyRef() {
-// 		return nil, errResolve
-// 	}
-
-// 	valueFromSecret, err := secretref.GetValue(ctx, c, *value.ValueFrom.SecretKeyRef)
-// 	if err != nil {
-// 		return nil, errResolve
-// 	}
-
-// 	return valueFromSecret, nil
-// }
-
-func resolveValue(ctx context.Context, c client.Reader, value telemetryv1alpha1.ValueType, errResolve error) ([]byte, error) {
+func resolveValue(ctx context.Context, c client.Reader, value telemetryv1alpha1.ValueType) ([]byte, error) {
 	if value.Value != "" {
 		return []byte(value.Value), nil
 	}

--- a/internal/tlscert/tls_cert_validator.go
+++ b/internal/tlscert/tls_cert_validator.go
@@ -109,7 +109,7 @@ func (v *Validator) Validate(ctx context.Context, config TLSBundle) error {
 		return err
 	}
 
-	// CA is optional
+	// CA is optional (check if missing)
 	missingCA := false
 	caPEM, err := resolveValue(ctx, v.client, *config.CA)
 	if err != nil {
@@ -132,17 +132,19 @@ func (v *Validator) Validate(ctx context.Context, config TLSBundle) error {
 		return err
 	}
 
-	// Parse the CA
-	var parsedCAs []*x509.Certificate
-	if !missingCA {
-		parsedCAs, err = parseCertificates(sanitizedCA, ErrCADecodeFailed, ErrCAParseFailed)
-		if err != nil {
-			return err
-		}
-	}
-
 	// Validate certificate
 	if err := validateCertificate(parsedCert, sanitizedCert, sanitizedKey, v.now()); err != nil {
+		return err
+	}
+
+	// CA is optional (don't parse and validate if missing)
+	if missingCA {
+		return nil
+	}
+
+	// Parse the CA
+	parsedCAs, err := parseCertificates(sanitizedCA, ErrCADecodeFailed, ErrCAParseFailed)
+	if err != nil {
 		return err
 	}
 

--- a/internal/tlscert/tls_cert_validator.go
+++ b/internal/tlscert/tls_cert_validator.go
@@ -136,26 +136,26 @@ func (v *Validator) Validate(ctx context.Context, config TLSBundle) error {
 		return err
 	}
 
+	// Parse the CA(s)
+	var parsedCAs []*x509.Certificate
+	if !missingCA {
+		parsedCAs, err = parseCertificates(sanitizedCA, ErrCADecodeFailed, ErrCAParseFailed)
+		if err != nil {
+			return err
+		}
+	}
+
 	// Validate certificate
 	if err := validateCertificate(parsedCert, sanitizedCert, sanitizedKey, v.now()); err != nil {
 		return err
 	}
 
-	// CA is optional (don't parse and validate if missing)
-	if missingCA {
-		return nil
-	}
-
-	// Parse the CA(s)
-	parsedCAs, err := parseCertificates(sanitizedCA, ErrCADecodeFailed, ErrCAParseFailed)
-	if err != nil {
-		return err
-	}
-
 	// Validate CA(s)
-	for _, ca := range parsedCAs {
-		if err := validateCA(ca, v.now()); err != nil {
-			return err
+	if !missingCA {
+		for _, ca := range parsedCAs {
+			if err := validateCA(ca, v.now()); err != nil {
+				return err
+			}
 		}
 	}
 

--- a/internal/tlscert/tls_cert_validator.go
+++ b/internal/tlscert/tls_cert_validator.go
@@ -99,20 +99,17 @@ func New(client client.Client) *Validator {
 }
 
 func (v *Validator) Validate(ctx context.Context, config TLSBundle) error {
-	certPEM, err := resolveValue(ctx, v.client, *config.Cert)
+	certPEM, err := resolveValue(ctx, v.client, *config.Cert, ErrValueResolveFailed)
 	if err != nil {
 		return err
 	}
 
-	keyPEM, err := resolveValue(ctx, v.client, *config.Key)
+	keyPEM, err := resolveValue(ctx, v.client, *config.Key, ErrValueResolveFailed)
 	if err != nil {
 		return err
 	}
 
-	caPEM, err := resolveValue(ctx, v.client, *config.CA)
-	if err != nil {
-		return err
-	}
+	caPEM, _ := resolveValue(ctx, v.client, *config.CA, nil)
 
 	// Make the best effort replacement of linebreaks in cert/key if present.
 	sanitizedCert := bytes.ReplaceAll(certPEM, []byte("\\n"), []byte("\n"))
@@ -131,9 +128,12 @@ func (v *Validator) Validate(ctx context.Context, config TLSBundle) error {
 	}
 
 	// Parse the CA
-	parsedCAs, err := parseCertificates(sanitizedCA, ErrCADecodeFailed, ErrCAParseFailed)
-	if err != nil {
-		return err
+	var parsedCAs []*x509.Certificate
+	if caPEM != nil {
+		parsedCAs, err = parseCertificates(sanitizedCA, ErrCADecodeFailed, ErrCAParseFailed)
+		if err != nil {
+			return err
+		}
 	}
 
 	// Validate certificate
@@ -232,7 +232,24 @@ func validateCA(ca *x509.Certificate, now time.Time) error {
 	return nil
 }
 
-func resolveValue(ctx context.Context, c client.Reader, value telemetryv1alpha1.ValueType) ([]byte, error) {
+// func resolveValue(ctx context.Context, c client.Reader, value telemetryv1alpha1.ValueType, errResolve error) ([]byte, error) {
+// 	if value.Value != "" {
+// 		return []byte(value.Value), nil
+// 	}
+
+// 	if value.ValueFrom == nil || !value.ValueFrom.IsSecretKeyRef() {
+// 		return nil, errResolve
+// 	}
+
+// 	valueFromSecret, err := secretref.GetValue(ctx, c, *value.ValueFrom.SecretKeyRef)
+// 	if err != nil {
+// 		return nil, errResolve
+// 	}
+
+// 	return valueFromSecret, nil
+// }
+
+func resolveValue(ctx context.Context, c client.Reader, value telemetryv1alpha1.ValueType, errResolve error) ([]byte, error) {
 	if value.Value != "" {
 		return []byte(value.Value), nil
 	}

--- a/internal/tlscert/tls_cert_validator_test.go
+++ b/internal/tlscert/tls_cert_validator_test.go
@@ -556,6 +556,25 @@ func TestResolveValue(t *testing.T) {
 			expectedErr: ErrCertDecodeFailed,
 		},
 		{
+			name: "ca empty",
+			inputCert: telemetryv1alpha1.ValueType{ValueFrom: &telemetryv1alpha1.ValueFromSource{
+				SecretKeyRef: &telemetryv1alpha1.SecretKeyRef{
+					Name:      "test",
+					Namespace: "default",
+					Key:       "cert",
+				}},
+			},
+			inputKey: telemetryv1alpha1.ValueType{ValueFrom: &telemetryv1alpha1.ValueFromSource{
+				SecretKeyRef: &telemetryv1alpha1.SecretKeyRef{
+					Name:      "test",
+					Namespace: "default",
+					Key:       "key",
+				}},
+			},
+			inputCa:     telemetryv1alpha1.ValueType{Value: ""},
+			expectedErr: ErrCertDecodeFailed,
+		},
+		{
 			name: "certs and key are present",
 			inputCert: telemetryv1alpha1.ValueType{ValueFrom: &telemetryv1alpha1.ValueFromSource{
 				SecretKeyRef: &telemetryv1alpha1.SecretKeyRef{

--- a/internal/tlscert/tls_cert_validator_test.go
+++ b/internal/tlscert/tls_cert_validator_test.go
@@ -407,7 +407,7 @@ rZ3xPLf7G+fObmeO7XuIoDfJHH6HDrdhhWi3F918KQ==
 	require.NoError(t, err)
 }
 
-func TestMissingCA(t *testing.T) {
+func TestEmptyCA(t *testing.T) {
 
 	oneMonthBeforeExpiry := pastCaExpiry.Add(-30 * 24 * time.Hour)
 	fakeClient := fake.NewClientBuilder().Build()

--- a/internal/tlscert/tls_cert_validator_test.go
+++ b/internal/tlscert/tls_cert_validator_test.go
@@ -407,6 +407,19 @@ rZ3xPLf7G+fObmeO7XuIoDfJHH6HDrdhhWi3F918KQ==
 	require.NoError(t, err)
 }
 
+func TestMissingCA(t *testing.T) {
+
+	oneMonthBeforeExpiry := pastCaExpiry.Add(-30 * 24 * time.Hour)
+	fakeClient := fake.NewClientBuilder().Build()
+	validator := Validator{
+		client: fakeClient,
+		now:    func() time.Time { return oneMonthBeforeExpiry },
+	}
+
+	err := validator.Validate(context.Background(), getTLSConfig(defaultCertData, defaultKeyData, []byte(``)))
+	require.NoError(t, err)
+}
+
 func TestSanitizeTLSSecretWithEscapedNewLine(t *testing.T) {
 	certData := "-----BEGIN CERTIFICATE-----\\nMIICgTCCAeqgAwIBAgIUcejYeiQzytrNkJ9G+V9mxRYZ4rQwDQYJKoZIhvcNAQEL\nBQAwXjELMAkGA1UEBhMCVVMxCzAJBgNVBAgMAk5ZMREwDwYDVQQHDAhOZXcgWW9y\nazEdMBsGA1UECgwUTW9jayBDQSBPcmdhbml6YXRpb24xEDAOBgNVBAMMB01vY2sg\nQ0EwIBcNMjQwMjE5MTQyNDE0WhgPNTAyNDAzMTkxNDI0MTRaMGAxCzAJBgNVBAYT\nAlVTMQswCQYDVQQIDAJOWTERMA8GA1UEBwwITmV3IFlvcmsxGjAYBgNVBAoMEU1v\nY2sgT3JnYW5pemF0aW9uMRUwEwYDVQQDDAx3d3cubW9jay5jb20wgZ8wDQYJKoZI\nhvcNAQEBBQADgY0AMIGJAoGBAN2iJZKzxmxXE8fzw5L45N1xqK1+CTCt5j12fJRT\nzruRxPH5IG1XOiN0vkJUmac3E7o0nsrUaxwP+kb68zZU4mExFVYTI1aVZHQdLXyC\noNBxAtk6Fy9P5fLfRFcovf4t8Frfdn8B4uCwBv5ywV9519kvKUdh0cfW8ZxSXloS\n4hdzAgMBAAGjODA2MAwGA1UdEwEB/wQCMAAwDgYDVR0PAQH/BAQDAgWgMBYGA1Ud\nJQEB/wQMMAoGCCsGAQUFBwMBMA0GCSqGSIb3DQEBCwUAA4GBAJVp4gGV2o1ZHSoR\nrS+Yq4u3eOqA6/nZtKzXq5cSfD4gyzkK/KnFxWZDElMPOrbXUtMVZvoxdyBbmrI8\nPsKzeS+TESxjIQvtIUYxZgbIDePQcXsJSpAjb+QoV0xOonfRtgKTd+s+/KNohDWs\nGWyZ1VBE/Yt3zH4DHhwXChzhUco7\\n-----END CERTIFICATE-----\n"
 	keyData := "-----BEGIN PRIVATE KEY-----\\nMIICXQIBAAKBgQDdoiWSs8ZsVxPH88OS+OTdcaitfgkwreY9dnyUU867kcTx+SBt\nVzojdL5CVJmnNxO6NJ7K1GscD/pG+vM2VOJhMRVWEyNWlWR0HS18gqDQcQLZOhcv\nT+Xy30RXKL3+LfBa33Z/AeLgsAb+csFfedfZLylHYdHH1vGcUl5aEuIXcwIDAQAB\nAoGANts0S4w9l4Ex/zKhfJYoJ3tDUbW5VpgkPaA/E4NuztQ0l+OemBGX7UCu+sHv\nygiC1HrDttY+sJJv0vO4EQGPiiLiGzgcuMV96wmRIHw1C3AJvZ9m71/ZoWb9ijIy\nGxe7SBD6Vw2Whyl3K25Wu9ZGAdKFeCbTeY0Dlm7lXw2Ch6kCQQD1QPDxdGjWIaru\n6DX6OwE+IlqwPwb9IdKpKdJzMYLnKmFxfLH5/LxzpKZrxsMZQ+PAFySQWc/ZVXpv\np9QhXLhvAkEA51hAi6D+uaW7EGB7nkNg7bOZjhS3dCkohMcfkQI19uUJWXv8eQPp\nDa4RTM/OqJwJoGKQ+NW5Q+aZwhQwtOirPQJBANMI64tJWQCQ/e4PwIquhTY7B4BK\n66+bkBLiGuXmf7Z8oFawLtFmqZ502oM5CB5QbcSX5W2U6qYfyHgVmRKQH18CQQDa\nrPT2BwxAd4PHCyxOgOoSRf4T60ktp+oA+CfCbhCMfBrGVwhja2rT34HC1XtGrZf7\n3q+iRoOEx2j3pxYTKwsRAkBG4hHXdRZ1a5RnTKi+qeVx5WPtOGuJa1hezGFs6l2p\nstp1Bpl+MQVzbPrhmc2O21g1CRhHDw75iRieVj04dYPP\\n-----END PRIVATE KEY-----\n"
@@ -440,6 +453,7 @@ func TestResolveValue(t *testing.T) {
 		Data: map[string][]byte{
 			"cert": []byte("cert"),
 			"key":  []byte("key"),
+			"ca":   []byte("ca"),
 		},
 	}
 
@@ -473,7 +487,7 @@ func TestResolveValue(t *testing.T) {
 				SecretKeyRef: &telemetryv1alpha1.SecretKeyRef{
 					Name:      "test",
 					Namespace: "default",
-					Key:       "cert",
+					Key:       "ca",
 				}},
 			},
 			expectedErr: ErrValueResolveFailed,
@@ -498,7 +512,7 @@ func TestResolveValue(t *testing.T) {
 				SecretKeyRef: &telemetryv1alpha1.SecretKeyRef{
 					Name:      "test",
 					Namespace: "default",
-					Key:       "cert",
+					Key:       "ca",
 				}},
 			},
 			expectedErr: ErrValueResolveFailed,
@@ -523,10 +537,10 @@ func TestResolveValue(t *testing.T) {
 				SecretKeyRef: &telemetryv1alpha1.SecretKeyRef{
 					Name:      "unknown",
 					Namespace: "default",
-					Key:       "cert",
+					Key:       "ca",
 				}},
 			},
-			expectedErr: ErrValueResolveFailed,
+			expectedErr: ErrCertDecodeFailed,
 		},
 		{
 			name: "certs and key are present",
@@ -548,7 +562,7 @@ func TestResolveValue(t *testing.T) {
 				SecretKeyRef: &telemetryv1alpha1.SecretKeyRef{
 					Name:      "test",
 					Namespace: "default",
-					Key:       "cert",
+					Key:       "ca",
 				}},
 			},
 			expectedErr: ErrCertDecodeFailed,

--- a/internal/tlscert/tls_cert_validator_test.go
+++ b/internal/tlscert/tls_cert_validator_test.go
@@ -420,6 +420,19 @@ func TestMissingCA(t *testing.T) {
 	require.NoError(t, err)
 }
 
+func TestNilCA(t *testing.T) {
+
+	oneMonthBeforeExpiry := pastCaExpiry.Add(-30 * 24 * time.Hour)
+	fakeClient := fake.NewClientBuilder().Build()
+	validator := Validator{
+		client: fakeClient,
+		now:    func() time.Time { return oneMonthBeforeExpiry },
+	}
+
+	err := validator.Validate(context.Background(), getTLSConfig(defaultCertData, defaultKeyData, nil))
+	require.NoError(t, err)
+}
+
 func TestSanitizeTLSSecretWithEscapedNewLine(t *testing.T) {
 	certData := "-----BEGIN CERTIFICATE-----\\nMIICgTCCAeqgAwIBAgIUcejYeiQzytrNkJ9G+V9mxRYZ4rQwDQYJKoZIhvcNAQEL\nBQAwXjELMAkGA1UEBhMCVVMxCzAJBgNVBAgMAk5ZMREwDwYDVQQHDAhOZXcgWW9y\nazEdMBsGA1UECgwUTW9jayBDQSBPcmdhbml6YXRpb24xEDAOBgNVBAMMB01vY2sg\nQ0EwIBcNMjQwMjE5MTQyNDE0WhgPNTAyNDAzMTkxNDI0MTRaMGAxCzAJBgNVBAYT\nAlVTMQswCQYDVQQIDAJOWTERMA8GA1UEBwwITmV3IFlvcmsxGjAYBgNVBAoMEU1v\nY2sgT3JnYW5pemF0aW9uMRUwEwYDVQQDDAx3d3cubW9jay5jb20wgZ8wDQYJKoZI\nhvcNAQEBBQADgY0AMIGJAoGBAN2iJZKzxmxXE8fzw5L45N1xqK1+CTCt5j12fJRT\nzruRxPH5IG1XOiN0vkJUmac3E7o0nsrUaxwP+kb68zZU4mExFVYTI1aVZHQdLXyC\noNBxAtk6Fy9P5fLfRFcovf4t8Frfdn8B4uCwBv5ywV9519kvKUdh0cfW8ZxSXloS\n4hdzAgMBAAGjODA2MAwGA1UdEwEB/wQCMAAwDgYDVR0PAQH/BAQDAgWgMBYGA1Ud\nJQEB/wQMMAoGCCsGAQUFBwMBMA0GCSqGSIb3DQEBCwUAA4GBAJVp4gGV2o1ZHSoR\nrS+Yq4u3eOqA6/nZtKzXq5cSfD4gyzkK/KnFxWZDElMPOrbXUtMVZvoxdyBbmrI8\nPsKzeS+TESxjIQvtIUYxZgbIDePQcXsJSpAjb+QoV0xOonfRtgKTd+s+/KNohDWs\nGWyZ1VBE/Yt3zH4DHhwXChzhUco7\\n-----END CERTIFICATE-----\n"
 	keyData := "-----BEGIN PRIVATE KEY-----\\nMIICXQIBAAKBgQDdoiWSs8ZsVxPH88OS+OTdcaitfgkwreY9dnyUU867kcTx+SBt\nVzojdL5CVJmnNxO6NJ7K1GscD/pG+vM2VOJhMRVWEyNWlWR0HS18gqDQcQLZOhcv\nT+Xy30RXKL3+LfBa33Z/AeLgsAb+csFfedfZLylHYdHH1vGcUl5aEuIXcwIDAQAB\nAoGANts0S4w9l4Ex/zKhfJYoJ3tDUbW5VpgkPaA/E4NuztQ0l+OemBGX7UCu+sHv\nygiC1HrDttY+sJJv0vO4EQGPiiLiGzgcuMV96wmRIHw1C3AJvZ9m71/ZoWb9ijIy\nGxe7SBD6Vw2Whyl3K25Wu9ZGAdKFeCbTeY0Dlm7lXw2Ch6kCQQD1QPDxdGjWIaru\n6DX6OwE+IlqwPwb9IdKpKdJzMYLnKmFxfLH5/LxzpKZrxsMZQ+PAFySQWc/ZVXpv\np9QhXLhvAkEA51hAi6D+uaW7EGB7nkNg7bOZjhS3dCkohMcfkQI19uUJWXv8eQPp\nDa4RTM/OqJwJoGKQ+NW5Q+aZwhQwtOirPQJBANMI64tJWQCQ/e4PwIquhTY7B4BK\n66+bkBLiGuXmf7Z8oFawLtFmqZ502oM5CB5QbcSX5W2U6qYfyHgVmRKQH18CQQDa\nrPT2BwxAd4PHCyxOgOoSRf4T60ktp+oA+CfCbhCMfBrGVwhja2rT34HC1XtGrZf7\n3q+iRoOEx2j3pxYTKwsRAkBG4hHXdRZ1a5RnTKi+qeVx5WPtOGuJa1hezGFs6l2p\nstp1Bpl+MQVzbPrhmc2O21g1CRhHDw75iRieVj04dYPP\\n-----END PRIVATE KEY-----\n"

--- a/internal/tlscert/tls_cert_validator_test.go
+++ b/internal/tlscert/tls_cert_validator_test.go
@@ -417,7 +417,7 @@ func TestEmptyCA(t *testing.T) {
 	}
 
 	err := validator.Validate(context.Background(), getTLSConfig(defaultCertData, defaultKeyData, []byte(``)))
-	require.NoError(t, err)
+	require.ErrorIs(t, err, ErrValueResolveFailed)
 }
 
 func TestNilCA(t *testing.T) {
@@ -430,7 +430,7 @@ func TestNilCA(t *testing.T) {
 	}
 
 	err := validator.Validate(context.Background(), getTLSConfig(defaultCertData, defaultKeyData, nil))
-	require.NoError(t, err)
+	require.ErrorIs(t, err, ErrValueResolveFailed)
 }
 
 func TestSanitizeTLSSecretWithEscapedNewLine(t *testing.T) {
@@ -553,7 +553,7 @@ func TestResolveValue(t *testing.T) {
 					Key:       "ca",
 				}},
 			},
-			expectedErr: ErrCertDecodeFailed,
+			expectedErr: ErrValueResolveFailed,
 		},
 		{
 			name: "ca empty",
@@ -572,7 +572,7 @@ func TestResolveValue(t *testing.T) {
 				}},
 			},
 			inputCa:     telemetryv1alpha1.ValueType{Value: ""},
-			expectedErr: ErrCertDecodeFailed,
+			expectedErr: ErrValueResolveFailed,
 		},
 		{
 			name: "certs and key are present",

